packages/patternfly-4/react-core/src/components/TextInput/examples/TextInput.md

```diff
@@ -79,3 +79,19 @@ class InvalidTextInput extends React.Component {
   }
 }
 ```
+
+```js title=Focus-using-ref
+import React from 'react';
+import { TextInput, Button } from '@patternfly/react-core';
+
+TextInputFocus = () => {
+  const [value, setValue] = React.useState('');
+  const ref = React.useRef(null);
+  return (
+    <>
+      <TextInput ref={ref} value={value} onChange={value => setValue(value)} aria-label="focused" />
+      <Button onClick={() => ref && ref.current && ref.current.focus()}>Focus</Button>
```

Comment:
Instead of a primary button, thinking we should follow the select example and use a checkbox?

https://patternfly-react.surge.sh/patternfly-4/documentation/react/components/select

Reply:
Checkbox for what? ref and refless?

Reply:
The button is used to set focus, but I don't know that we should use primary buttons in our examples like this. 

We need to be careful about the use cases we demo; otherwise, folks may use this incorrectly. I don't believe that using a primary button here is a valid pattern.

I didn't go thru all the examples, but the "select" example uses checkboxes to show features. I'm suggesting we use that as precedence.

Reply:
Hi @dlabrecq - I updated the example to something much simpler. This one doesn't require checkbox/radio/button to demonstrate "ref".